### ISSUE TITLE
Dockerfile using Multi-build

### ISSUE THREAD

**Issue by pascalandy** [docker-library/ghost#95]
Hey folks!

I was looking to slim the Ghost image using multi-build and it looks someone [made it happen](https://blog.mornati.net/docker-ghost-blog-slim-down/). Wondering if there is a plan to update the official dockerfile and use multi-build? 

I understand that some folks would prefer the actual image for developing while most of us want to use the image no-question-ask. Maybe having two versions would make sense?

Cheers!
Pascal

**Comment by yosifkit** [comment]
1. Official images do not support multi-stage builds; see https://github.com/docker-library/official-images/issues/3383
1. The image does not install anything that is not required by ghost-cli and ghost
   - perhaps a minimal version that installs `ghost-cli` to install ghost and then removes ghost-cli, all in one layer, but that would leave out the database initialization as well (`knex-migrator-migrate --init --mgpath "$GHOST_INSTALL/current"`).

@acburdine, what do you think about an image without the `ghost-cli` or `knex-migrator`?  Would there be value in that?

**Comment by pascalandy** [comment]
Closing as the maintainer will take action once Docker EE support multi-build.

**Comment by acburdine** [comment]
Thinking about it, it should be possible to remove the ghost-cli dependency. There was a change in Ghost-CLI recently too that removed the need to globally install knex-migrator - it now uses the one that's installed with Ghost itself.

tl;dr it should be too difficult.

**Comment by pascalandy** [comment]
Meanwhile, I forked the official project here. It supports the multi-stage build : 
https://github.com/firepress-org/ghostfire

There are other features as well. Check out the README. 

Cheers!
P
